drivers/sensor/liteon/ltr329/ltr329.c

```diff
@@ -0,0 +1,363 @@
+}
+
+static int ltr329_read_register(const struct i2c_dt_spec *bus, uint8_t reg, uint8_t *value)
+{
+	return i2c_write_read_dt(bus, &reg, sizeof(reg), value, sizeof(*value));
+}
+
+static int ltr329_check_device_id(const struct i2c_dt_spec *bus)
+{
+	uint8_t id;
+	int rc;
+
+	rc = ltr329_read_register(bus, LTR329_PART_ID, &id);
+	if (rc < 0) {
+		LOG_ERR("Failed to read PART_ID");
+		return rc;
+	}
+	if (id != LTR329_PART_ID_VALUE) {
+		LOG_ERR("PART_ID mismatch: expected 0x%02X, got 0x%02X", LTR329_PART_ID_VALUE, id);
+		return -ENODEV;
+	}
+
+	rc = ltr329_read_register(bus, LTR329_MANUFAC_ID, &id);
+	if (rc < 0) {
+		LOG_ERR("Failed to read MANUFAC_ID");
+		return rc;
+	}
+	if (id != LTR329_MANUFACTURER_ID_VALUE) {
+		LOG_ERR("MANUFAC_ID mismatch: expected 0x%02X, got 0x%02X",
+			LTR329_MANUFACTURER_ID_VALUE, id);
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int ltr329_init_registers(const struct i2c_dt_spec *bus, const struct ltr329_config *cfg)
+{
+	const uint8_t control_reg = LTR329_REG_SET(LTR329_ALS_CONTR, MODE, 1) |
+				    LTR329_REG_SET(LTR329_ALS_CONTR, GAIN, cfg->gain);
+	const uint8_t meas_reg = LTR329_REG_SET(LTR329_MEAS_RATE, REPEAT, cfg->measurement_rate) |
+				 LTR329_REG_SET(LTR329_MEAS_RATE, INT_TIME, cfg->integration_time);
+	uint8_t buffer;
+	int rc;
+
+	rc = ltr329_send_command(bus, LTR329_ALS_CONTR, control_reg);
+	if (rc < 0) {
+		LOG_ERR("Failed to set ALS_CONTR register");
+		return rc;
+	}
+
+	rc = ltr329_send_command(bus, LTR329_MEAS_RATE, meas_reg);
+	if (rc < 0) {
+		LOG_ERR("Failed to set MEAS_RATE register");
+		return rc;
+	}
+
+	/* Read back the MEAS_RATE register to verify the settings */
+	rc = ltr329_read_register(bus, LTR329_MEAS_RATE, &buffer);
+	if (rc < 0) {
+		LOG_ERR("Failed to read back MEAS_RATE register");
+		return rc;
+	}
+	if (LTR329_REG_GET(LTR329_MEAS_RATE, REPEAT, buffer) != cfg->measurement_rate) {
+		LOG_ERR("Measurement rate mismatch: expected %u, got %u", cfg->measurement_rate,
+			(uint8_t)LTR329_REG_GET(LTR329_MEAS_RATE, REPEAT, buffer));
+		return -ENODEV;
+	}
+	if (LTR329_REG_GET(LTR329_MEAS_RATE, INT_TIME, buffer) != cfg->integration_time) {
+		LOG_ERR("Integration time mismatch: expected %u, got %u", cfg->integration_time,
+			(uint8_t)LTR329_REG_GET(LTR329_MEAS_RATE, INT_TIME, buffer));
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+static int ltr329_init(const struct device *dev)
+{
+	const struct ltr329_config *cfg = dev->config;
+	int rc;
+
+	if (!i2c_is_ready_dt(&cfg->bus)) {
+		LOG_ERR("I2C bus not ready");
+		return -ENODEV;
+	}
+
+	/* Wait for sensor startup */
+	k_sleep(K_MSEC(LTR329_INIT_STARTUP_MS));
+
+	rc = ltr329_check_device_id(&cfg->bus);
+	if (rc < 0) {
+		return rc;
+	}
+
+	/* Init register to enable sensor to active mode */
+	rc = ltr329_init_registers(&cfg->bus, cfg);
+	if (rc < 0) {
+		return rc;
+	}
+
+	/* Wait for sensor to wake up from standby - this time is required by datasheet */
+	k_sleep(K_MSEC(LTR329_WAKEUP_FROM_STANDBY_MS));
```

Comment:
Do we really need to sleep here right before just returning 0? I think if the app calls fetch too soon you'll catch it with the `ltr329_check_data_ready`